mnz-client/internal/mnz/mnz.go

```diff
@@ -0,0 +1,170 @@
+package mnz
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/go-retryablehttp"
+	"io"
+	"strings"
+	"time"
+)
+
+type ArgType struct {
+	Name           string                 `json:"name"`
+	AvailableSpecs map[string]interface{} `json:"-"`
+	RequiredSpecs  []string               `json:"-"`
+}
+
+type Arg struct {
+	Name    string         `json:"-"`
+	ArgType ArgType        `json:"argType"`
+	Specs   map[string]any `json:"specs"`
+}
+
+func getArgType(s string) (ArgType, error) {
+	switch s {
+	case "file":
+		return FileArgType, nil
+	}
+	return ArgType{}, errors.New("unknown arg type")
+}
+
+type RunSpecRequest struct {
+	License     string         `json:"license"`
+	ProductName string         `json:"productName"`
+	RunSpec     map[string]Arg `json:"runSpec"`
```

Comment:
Just a note: marshalling this to json produces something like:
```json
"runSpec": {
  "a": {
    "argType": {"name": "file"},
    "specs": {...}
  }
}
```

If this is what you want, it's OK. But knowing how type discriminators work in JS, I expect this
```json
"runSpec": {
  "a": {
    "argType": "file",
    "specs": {...}
  }
}
```
to be more natural and comfortable when working with requests on server side.

I do not know the actual API structure, just got an insight this structure might be unintentional.